cpp/src/arrow/compute/kernels/vector_demean.cc

```diff
@@ -0,0 +1,81 @@
+
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// Include necessary Arrow headers
+#include <arrow/compute/api_vector.h>
+#include "arrow/compute/api_scalar.h"
+#include "arrow/compute/function.h"
+#include "arrow/compute/registry.h"
+#include "arrow/util/logging.h"
+
+namespace arrow {
+namespace compute {
+namespace internal {
+
+namespace {
+// Function documentation
+const FunctionDoc demean_doc{
+    "Perform a demean operation over all the elements of the array",
+    ("Returns an array where every element has been removed from \n"
+     "the mean of the array."),
+    {"array"}};
+
+class DemeanMetaFunction : public MetaFunction {
+ public:
+  DemeanMetaFunction() : MetaFunction("demean", Arity::Unary(), demean_doc) {}
+
+  Result<Datum> ExecuteImpl(const std::vector<Datum>& args,
+                            const FunctionOptions* options,
+                            ExecContext* ctx) const override {
+    switch (args[0].kind()) {
+      case Datum::ARRAY:
+      case Datum::CHUNKED_ARRAY: {
```

Comment:
I would say support for `chunked_array` is not needed - more of a "nice to have". The main application of these kernels are to be used with `Acero` which doesn't use chunk arrays.